rpc/core/types/responses.go

```diff
@@ -104,18 +104,18 @@ type ResultBroadcastTx struct {
 }
 
 type ResultBroadcastTxCommit struct {
-	CheckTx   abci.Result `json:"check_tx"`
-	DeliverTx abci.Result `json:"deliver_tx"`
-	Hash      data.Bytes  `json:"hash"`
-	Height    int         `json:"height"`
+	CheckTx   abci.ResponseCheckTx   `json:"check_tx"`
+	DeliverTx abci.ResponseDeliverTx `json:"deliver_tx"`
+	Hash      data.Bytes             `json:"hash"`
+	Height    uint64                 `json:"height"`
 }
 
 type ResultTx struct {
-	Height   int           `json:"height"`
-	Index    int           `json:"index"`
-	TxResult abci.Result   `json:"tx_result"`
-	Tx       types.Tx      `json:"tx"`
-	Proof    types.TxProof `json:"proof,omitempty"`
+	Height   uint64                 `json:"height"`
+	Index    uint32                 `json:"index"`
+	TxResult abci.ResponseDeliverTx `json:"tx_result"`
+	Tx       types.Tx               `json:"tx"`
```

Comment:
just a note that this is rendered in base64 since it doesn't use data.Bytes. Will probably annoy someone some day ...

Reply:
Could you please remind me why are storing tx itself? like we should already have it in the block. Most clients will want the receipt (result), not the original tx anyway. see https://github.com/tendermint/tendermint/issues/905

Reply:
we do need some way to get the tx we want without fetching the whole block. I like #905. But maybe we should wait until after 0.13 is out to address it.